core/src/main/java/com/linecorp/armeria/common/logging/ContentPreviewerFactory.java

```diff
@@ -0,0 +1,151 @@
+        if (factory2 == DISABLED) {
+            return factory1;
+        }
+        if (factory1 instanceof ContentPreviewerFactoryCombinable) {
+            return ((ContentPreviewerFactoryCombinable) factory1).combine(factory2);
+        }
+        if (factory2 instanceof ContentPreviewerFactoryCombinable) {
+            return ((ContentPreviewerFactoryCombinable) factory2).combine(factory1);
+        }
+        return new CompositeContentPreviewerFactory(Arrays.asList(factory1, factory2));
+    }
+
+    /**
+     * TODO: AddJavadocs.
+     */
+    static ContentPreviewerFactory of(ContentPreviewerFactory... factories) {
+        final List<ContentPreviewerFactory> factoryList = new ArrayList<>();
+        ContentPreviewerFactory combined = DISABLED;
+        for (ContentPreviewerFactory factory : factories) {
+            if (factory == DISABLED) {
+                continue;
+            }
+            if (factory instanceof ContentPreviewerFactoryCombinable) {
+                combined = ((ContentPreviewerFactoryCombinable) factory).combine(combined);
+            } else if (combined instanceof ContentPreviewerFactoryCombinable) {
+                combined = ((ContentPreviewerFactoryCombinable) combined).combine(factory);
+            }
+            factoryList.add(factory);
+        }
+        if (combined != DISABLED) {
+            factoryList.add(combined);
+        }
+        if (factoryList.size() < 2) {
+            return factoryList.isEmpty() ? DISABLED : factoryList.get(0);
+        }
+        return new CompositeContentPreviewerFactory(factoryList);
+    }
+
+    /**
+     * TODO: AddJavadocs.
+     */
+    static ContentPreviewerFactory of(Map<MediaType, Supplier<ContentPreviewer>> map) {
+        return new MappedContentPreviewerFactory(map);
+    }
+
+    /**
+     * TODO: AddJavadocs.
+     */
+    static ContentPreviewerFactory of(Supplier<ContentPreviewer> supplier, MediaType... contentTypes) {
+        final Map<MediaType, Supplier<ContentPreviewer>> maps = new HashMap<>();
+        for (MediaType type : contentTypes) {
+            maps.put(type, supplier);
+        }
+        return of(maps);
+    }
+
+    static ContentPreviewerFactory of(Supplier<ContentPreviewer> supplier, String... contentTypes) {
+        return of(supplier, Arrays.stream(contentTypes).map(MediaType::parse).toArray(MediaType[]::new));
+    }
+
+    static ContentPreviewerFactory ofString(int length, Charset defaultCharset, MediaType... contentTypes) {
+        return of(() -> ContentPreviewer.ofString(length, defaultCharset), contentTypes);
+    }
+
+    static ContentPreviewerFactory ofString(int length, Charset defaultCharset, String... contentTypes) {
+        return of(() -> ContentPreviewer.ofString(length, defaultCharset), contentTypes);
+    }
+
+    static ContentPreviewerFactory ofString(int length, Charset defaultCharset) {
+        return ofString(length, defaultCharset, MediaType.ANY_TEXT_TYPE, MediaType.ANY_APPLICATION_TYPE);
```

Comment:
`ANY_APPLICATION_TYPE` sounds too broad. How about narrowing it down a little bit?